CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+
+If you want to add a native implementation to an existing component, you may be better off making a duplicate "experimental" component that contains the native implementation, as using a Native Module limits you to one slot, where an existing JS component may have multiple slots. Examples of duplicate native implementation include "experimental-avatar" and "experimental-native-button".
+
+An example of how we would create Shimmer with the compose framework if all the swift properties on the native control were tokens would look like so. This is how it's currently implemented
+
+```
+export type ShimmerTokens = {
+  // add the native swift properties here
+}
+// the native control on the inside needs to receive these
+export type NativeShimmerProps = ShimmerTokens & ViewProps;
+// the shimmer control only should get these via customization
+export type ShimmerProps = ViewProps
+// tokensThatAreAlsoProps is false and can be omitted because nothing is shared
+// type looks like this:
+props: ShimmerProps;
+slotProps: { root: NativeShimmerProps };
+tokens: ShimmerTokens;
+// buildProps would look like this:
+root: buildProps(tokens => ({ ...tokens }), [])
+// this is because the tokens aren't depending on any props
+// which means they only change when the theme is updated
+```
+
+If instead, some of the native swift properties were tokens AND props, it would look something like this:
+
+```
+const tokensThatAreAlsoProps: (keyof ShimmerTokens)[] = [/* list all the tokens here */];
+
+// in the body of the component then add
+tokenProps,
+
+// then buildProps becomes:
+  root: buildProps(tokens => ({...tokens}), tokenProps)
+```
+
+To add the actual native module that wraps the FluentUI Apple control:
+
+1. Create a `podspec` file at the root of the your new component's directory.
+   - This is the file cocoapods will read to figure out what native files to import when it compiles the client app that is using your component
+   - We can read the metadata from our `package.json` (See the existing podspecs for how to do this)
+   - We specify "MicrosoftFluentUI" as dependency, which ensures the source code for FluentUI Apple is pulled in while compiling this component.
+1. Create the `ios/` and `macos/` subdirectories (Whichever platforms you need) in your components directory
+1. Inside each new directory you just created, you will have at a minimum of 3 files (2 if you skip Swift and only use objective c)
+   1. `<new-component>-Bridging-Header.h`
+      - This file allows your swift files to import/read React Native objective C headers. At a minimum, we need `<React/RCTViewManager.h>`
+   1. `<new-component>ViewManager.swift`
+      - This Swift file imports FluentUI Apple, and creates a subclass of RCTViewManager to instantiate and return your FluentUI Apple control.
+      - You can also overload the "constantsToExport" method to grab default values like colors and sizes to pass to JS.
```

Comment:
constantsToExport is an API so I'd use the code style highlighting like so- `constantsToExport`